ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,306 @@
+
+    for node in source.body:
+        if isinstance(node, (ast.Assign, ast.AnnAssign, ast.AugAssign)):
+            for target in node.targets:
+                if not isinstance(target, ast.Name):
+                    # ignore unpacked assignments
+                    continue
+                if target.id == "DOCUMENTATION":
+                    mod_map.doc_string = node.value.value
+                    mod_map.ds_line_start = node.lineno
+                    mod_map.ds_line_end = node.end_lineno
+                    break
+                elif target.id == "EXAMPLES":
+                    mod_map.example_string = node.value.value
+                    mod_map.ex_line_start = node.lineno
+                    mod_map.ex_line_end = node.end_lineno
+                    break
+        elif isinstance(node, ast.ClassDef):
+            class_map = ParsedClass(
+                name=node.name,
+                line_start=node.lineno,
+                line_end=node.end_lineno,
+            )
+            class_doc = ast.get_docstring(node, clean=False)
+            if class_doc is not None:
+                class_map.doc_string = class_doc
+                ds_span = re.search(re.escape(class_doc), content)
+                if ds_span is not None:
+                    class_map.ds_line_start = len(
+                        content[:ds_span.start()].splitlines()
+                    )
+                    class_map.ds_line_end = (
+                        class_map.ds_line_start + len(class_doc.splitlines())
+                    )
+
+            for child_node in ast.iter_child_nodes(node):
+                if isinstance(child_node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                    func_map = ParsedFunc(
+                        name=child_node.name,
+                        line_start=child_node.lineno,
+                        line_end=child_node.end_lineno,
+                    )
+                    func_doc = ast.get_docstring(child_node)
+                    if func_doc is not None:
+                        func_map.doc_string = func_doc
+                        ds_span = re.search(re.escape(func_doc), content)
+                        if ds_span is not None:
+                            func_map.ds_line_start = len(
+                                content[:ds_span.start()].splitlines()
+                            )
+                            func_map.ds_line_end = (
+                                func_map.ds_line_start + len(func_doc.splitlines())
+                        )
+                    class_map.funcs.append(func_map)
+
+            mod_map.classes.append(class_map)
+
+    return mod_map
+
+def _check_py_changes(file_content, diff):
+    """ Check a python file's changes to see if they're only docstring
+        changes.
+    """
+
+    diff = _get_diff_info(diff)
+    source = None
+
+    if file_content is not None:
+        source = _get_ast_info(file_content)
+
+    if source is not None:
```

Comment:
We could just invert the conditional and return early here.

Reply:
And we should `return False` if source is `None` instead of returning `None` (implicitly) in that case.